cromshell

```diff
@@ -649,127 +811,66 @@ function runSubCommandOnWorkflowId()
 # The bash equivalent to `if __name__ == '__main__':
 if ${ISCALLEDBYUSER} ; then
 
-  #Check given arguments:
-  if [[ $# -gt $MAXARGS ]] ; then
-    simpleUsage
-    exit 1
-  elif [[ $# -lt $MINARGS ]] ; then
-    simpleUsage
-    exit 2
-  fi
+  # Check the remaining arguments for help and display it if we have to: 
+  checkForComplexHelpArgument $@
 
-  #Read args:
-  while [ $# -gt 0 ] ; do
-
-    case "$1" in
-      -h|--h|--help|-help|help)
-        usage;
-        exit 0;
-        ;;
-      submit|status|logs|execution-status-count|metadata|slim-metadata|timing|abort|notify|list|fetch-all|fetch-logs|list-outputs) 
-        # Get our sub-command:
-        SUB_COMMAND=$1
-        shift
-        # We'll manually handle the rest of the options:
-        break
+  # Get top-level arguments to script itself:
+  while getopts "h" opt ; do
+    case ${opt} in
+      h) 
+        usage
+        exit 0
         ;;
       *)
-        error "${SCRIPTNAME}: invalid option: $1" 
         simpleUsage
-        echo "Try \`$SCRIPTNAME --help' for more information."
-        exit 3;
-        ;; 
+        exit 3
+        ;;
     esac
-
-    #Get next argument in $1:
-    shift
   done
+  shift $((OPTIND-1)) 
 
-  ################################################################################
+  # Get our sub-command:
+  SUB_COMMAND=${1}
+  shift
 
-  # Make sure we have all the required commands:
+  # Validate our sub-command:
+  case ${SUB_COMMAND} in
```

Comment:
additional help check, shouldn't that all be consolidated where you ran the advanced help check?